src/openni2/DeviceDriver.cpp

```diff
@@ -500,7 +500,43 @@ namespace Freenect2Driver
             WriteMessage("Opening device " + std::string(uri));
             int id = uri_to_devid(iter->first.uri);
             DeviceImpl* device = new DeviceImpl(id);
-            device->setFreenect2Device(freenect2.openDevice(id)); // XXX, detault pipeline // const PacketPipeline *factory);
+
+			const char *pipeline_type_env;
+			// GL, CL, CUDA, CPU
+			pipeline_type_env = std::getenv("LIBFREENECT2_NITE_PIPELINE_TYPE");
+			std::string pipeline_type;
+
+			if (pipeline_type_env)
+			{
+				pipeline_type = std::string(pipeline_type_env);
+			}
+
+			libfreenect2::PacketPipeline* pipeline = NULL;
+
+#if defined(LIBFREENECT2_WITH_OPENGL_SUPPORT)
+			if (pipeline_type == "GL")
+				pipeline = new libfreenect2::OpenGLPacketPipeline();
+#elif defined(LIBFREENECT2_WITH_CUDA_SUPPORT)
```

Comment:
You are right; this would be a bug -- it worked for me, but I had OpenCL pre-set so it did not show up and I can't test CUDA. I will fix it.